src/mono/mono/metadata/marshal.c

```diff
@@ -3940,10 +3960,55 @@ emit_managed_wrapper_noilgen (MonoMethodBuilder *mb, MonoMethodSignature *invoke
 }
 #endif
 
+static gboolean
+type_is_blittable (MonoType *type)
+{
+	if (type->byref)
+		return FALSE;
+	type = mono_type_get_underlying_type (type);
+	switch (type->type) {
+	case MONO_TYPE_I1:
+	case MONO_TYPE_U1:
+	case MONO_TYPE_I2:
+	case MONO_TYPE_U2:
+	case MONO_TYPE_I4:
+	case MONO_TYPE_U4:
+	case MONO_TYPE_I8:
+	case MONO_TYPE_U8:
+	case MONO_TYPE_R4:
+	case MONO_TYPE_R8:
+	case MONO_TYPE_I:
+	case MONO_TYPE_U:
+	case MONO_TYPE_CHAR:
```

Comment:
I think char shouldn't be here. Doesn't look like it's supposed to be blittable. (`mono_class_setup_mono_type` doesn't set the blittable bit on it)

Reply:
Ah yeah, good catch. Yes, it's definitely not for the same reason as strong—ambiguity over converting to ANSI, UTF-8, or UTF-16.